File: frontend/src/types/report.ts
export type ReportCategory =
  | 'STEP_FREE_ISSUE'
  | 'SURFACE_HAZARD'
  | 'OBSTRUCTION'
  | 'LIGHTING_ISSUE'
  | 'HEAT_NO_SHADE';

export type ReportSeverity = 'LOW' | 'MEDIUM' | 'HIGH';

export type ReportStatus = 'PENDING' | 'VERIFIED' | 'REJECTED' | 'RESOLVED' | 'EXPIRED';

export interface Report {
  reportId: string;
  reporterUserId: string;
  category: ReportCategory;
  severity: ReportSeverity;
  description: string;
  latitude: number;
  longitude: number;
  status: ReportStatus;
  confidenceScore: number;
  createdAt: string;
  expiresAt: string | null;
}

export interface NearbyReport {
  reportId: string;
  category: ReportCategory;
  severity: ReportSeverity;
  status: ReportStatus;
  latitude: number;
  longitude: number;
  confidenceScore: number;
  createdAt: string;
}

export interface DraftResponse {
  suggestedCategory: ReportCategory;
  suggestedSeverity: ReportSeverity;
  confidenceFlag: boolean;
  latitude: number;
  longitude: number;
  description: string;
}

export interface CreateReportRequest {
  description: string;
  latitude: number;
  longitude: number;
  category?: string;
  severity?: string;
}

File: backend/src/main/java/com/accesspath/models/Report.java
package com.accesspath.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.locationtech.jts.geom.Point;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "reports")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Report {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "report_id", updatable = false, nullable = false)
    private UUID reportId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reporter_user_id", nullable = false)
    private User reporter;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "last_updated_at")
    private Instant lastUpdatedAt;

    @Column(name = "expires_at")
    private Instant expiresAt;

    @Column(name = "status", nullable = false)
    @Enumerated(EnumType.STRING)
    private ReportStatus status;

    @Column(name = "category", nullable = false)
    private String category;

    @Column(name = "severity", nullable = false)
    private String severity;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "location_point", columnDefinition = "geometry(Point,4326)")
    private Point locationPoint;

    @Column(name = "confidence_score")
    private Float confidenceScore;

    public enum ReportStatus {
        PENDING,
        VERIFIED,
        REJECTED,
        RESOLVED,
        EXPIRED
    }
}